mtr.c

```diff
@@ -590,14 +590,13 @@ static void parse_arg (struct mtr_ctl *ctl, int argc, char **argv)
 }
 
 
-static void parse_mtr_options (struct mtr_ctl *ctl, char *string)
+static void parse_mtr_options (struct mtr_ctl *ctl, names_t **names, char *string)
 {
-  int argc;
+  int argc = 1;
   char *argv[128], *p;
 
   if (!string) return;
-
-  argv[0] = "mtr";
+  argv[0] = xstrdup(PACKAGE_NAME);
   argc = 1;
```

Comment:
I don't understand the question. This is again remedy to read-onlly data. If question is  static string from config.h rather than "mtr" then my answer is as simple as IMHO it is always best to avoid all magic strings and numbers, and prefer single symbolic definitions instead.